system/security/shared/caching.hpp

```diff
@@ -151,16 +151,25 @@ class CPermissionsCache
 
     virtual ~CPermissionsCache();
 
-    static CPermissionsCache* queryInstance()
+    //Returns a shared cache of a given class type.
+    //Call this method with a unique class string ("LDAP", "MyOtherSecMgr")
+    //to create a cache shared amongst security managers of the same class
+    static CPermissionsCache* queryInstance(const char * _secMgrClass)
     {
+        const char * secMgrClass = (_secMgrClass != nullptr  &&  *_secMgrClass) ? _secMgrClass : "genericSecMgrClass";
```

Comment:
I don't know if the "nameless" version is a common case and if you care about speed, but if the answer to both is "yes" you could consider bypassing the hash lookup for this case and just using a static variable to hold the nameless answer.